### DISCUSSION TITLE
Any gcc options to modify sizeof a type?

### DISCUSSION THREAD

**Discussion by rorySomething** [msys2/MINGW-packages#9829] [Q&A] [answered]
Good day,

I'm attempting to build Mit-scheme in MSYS2 with mingw.

When running make I very quickly hit an error.
`error: pointers must fit in 'unsigned long'`
because it fails this check
`if (SIZEOF_UINTPTR_T > SIZEOF_UNSIGNED_LONG)`

Are there any obscure compiler flags I can pass for this situation?
I have yet to try compiling with 32 bit mingw where I assume uintptr and unsigned long will both be 32 bits.

In the meantime I'm going to use guile that is available through pacman.  (Thank you to everyone involved.)

At some point I hope to contribute back to msys2 and mingw64.  I've been using it to try to make games with openGL that will be friendly to Linux and Windows if any are ever to a point of working and fun.

**Comment by mati865** [comment, marked as answer]
I'm not aware of such option and doubt somebody has created it.

On Windows `long` is 32-bit long so the correct fix is to use `long long` in your code.

**Reply by rorySomething** [reply]
Thank you, maybe I'll try to spend a few minutes getting crazy with sed.

Guile in the msys2 terminal is no fun with arrow keys or some other keyboard shortcut getting errors included in the entered command lines.
Endless errors with \x1b;  (Escape code? whenever I use arrow keys to fix text)

Eventually I'll cave and just install cygwin and run it there.

**Comment by jeremyd2019** [comment]
Cygwin is LP64 rather than LLP64.  https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models

**Reply by rorySomething** [reply]
But is mingw-gcc compiling for windows natively?  Not using cygwin emulation

**Reply by jeremyd2019** [reply]
right, which means it is LLP64.  If you want LP64 on Windows I think you need to use Cygwin

**Reply by rorySomething** [reply]
Thanks, I read that around the other way.
